homework-submission/abdullahReact/main.js

```diff
@@ -0,0 +1,98 @@
+const toDoItems = [
+  {
+    id: 1,
+    description: 'Get out of bed',
+    deadline: '2017-09-11',
+    done: true,
+  },
+  {
+    id: 2,
+    description: 'Brush teeth',
+    deadline: '2017-09-10',
+    done: false,
+  },
+  {
+    id: 3,
+    description: 'Eat breakfast',
+    deadline: '2017-09-09',
+    done: false,
+  },
+];
+
+//the goal of this component is to define/create a reusable list item that uses own function to set state.
+// in this case to set as done or not done by clicking
+
+class ListItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { done: props.done };
+    this.switchDoneState = this.switchDoneState.bind(this);
+  }
+
+  switchDoneState() {
+    this.setState({ done: !this.state.done });
+  }
+
+  render() {
+    const { description, deadline } = this.props;
+    return (
+      <li onClick={this.switchDoneState} className={`listItem ${this.state.done}`}>
+        {description} - {deadline}
+      </li>
+    );
+  }
+}
+
+//the goal of this component is to define/create a reusable list header.
+//it returns a header with taken parameter/prop.type + "List"
+
+const ListHeader = listHead => {
+  return <h2 className="listHeader">{listHead.type} List</h2>;
+};
+
+//the goal of this component is to create an unordered list statically
+//it returns a ul which has a header(with given prop.type) 3 item with given props.
+class StaticList extends React.Component {
+  render() {
+    return (
+      <ul>
+        <ListHeader type="Static" />
+        <ListItem description="Get out of bed" deadline="2017-09-11" />
+        <ListItem description="Brush teeth" deadline="2017-09-10" />
+        <ListItem description="Eat breakfast" deadline="2017-09-09" />
+      </ul>
+    );
+  }
+}
+//the goal of this component is to create an unordered list dynamically
+//it returns a ul which has items as many as given props.array has.
+class DynamicList extends React.Component {
+  render() {
+    const { toDoArr } = this.props;
+    console.log(this.props);
```

Comment:
Always good to remove `console.log`s before creating a pull request :)

Reply:
yes. I forgot it, now I removed it.